pkgs/sdk/vm/handler.go

```diff
@@ -193,6 +197,22 @@ func (vh vmHandler) queryFile(ctx sdk.Context, req abci.RequestQuery) (res abci.
 	return
 }
 
+func (vh vmHandler) queryFiles(ctx sdk.Context, req abci.RequestQuery) (res abci.ResponseQuery) {
```

Comment:
```suggestion
func (vh vmHandler) queryPkgFiles(ctx sdk.Context, req abci.RequestQuery) (res abci.ResponseQuery) {
```

Reply:
After more reflection, I think we should not have a `queryPkgFiles` handler that gets the bodies, but instead implement the `getPackage` helper returning a list of files with attributes and not only the body: https://github.com/gnolang/gno/pull/526/files#diff-86850325e38fa997f532d762282030b3e585caa785324e2187eceba36e8169d5R121

Reply:
To avoid going in the wrong direction, you want me to return the std.MemPackage directly?

```go
//keeper.go
func (vm *VMKeeper) QueryPackage(ctx sdk.Context, path string) (res std.MemPackage, err error) {
	dirpath, _ := std.SplitFilepath(path)
	store := vm.getGnoStore(ctx)
	if memPkg := store.GetMemPackage(dirpath); memPkg != nil {
		return memPkg, nil
	}

	return std.MemPackage{}, nil
}

//handler.go
func (vh vmHandler) queryPackage(ctx sdk.Context, req abci.RequestQuery) (res abci.ResponseQuery) {
	dirpath := string(req.Data)
	result, err := vh.vm.QueryPackage(ctx, dirpath)
	if err != nil {
		res = sdk.ABCIResponseQueryFromError(err)
		return
	}
	b, err := json.Marshal(&result)
	if err != nil {
		res = sdk.ABCIResponseQueryFromError(err)
		return
	}
	res.Data = b
	return 
}
```

Reply:
It may be beneficial to keep the API as concise as possible, rather than having numerous options that essentially perform the same task.